src/cswinrt/code_writers.h

```diff
@@ -2269,12 +2269,6 @@ private IObjectReference Make__%()
                                 bind<write_type_name>(semantics, typedef_name_type::StaticAbiClass, true)
                             );
                         }
-                        else
-                        {
-                            w.write(R"(global::System.Threading.Interlocked.CompareExchange(ref __%, (IObjectReference)typeof(IObjectReference).GetMethod("As", Type.EmptyTypes).MakeGenericMethod(typeof(%).GetHelperType().FindVftblType()).Invoke(((IWinRTObject)this).NativeObject, null), null);)",
```

Comment:
Mmh... How was this working at all on NAOT before? 🤔

Reply:
It probably wasn't, there was only like 2 classes that went into this block before.  Looking at if there is an easy test I can add for this scenario.

Reply:
Confirmed new code path we are now using for all generic types implemented by RCWs is already tested by the `DeviceInformation.FindAllAsync` tests which have an RCW type implementing a collection type.